From d345b36b483fec8fb2def5f3acdc2d3c5faa94eb Mon Sep 17 00:00:00 2001
From: Dan McGee <dan@archlinux.org>
Date: Sun, 19 Jun 2011 16:04:25 -0500
Subject: [PATCH] Set hostname via /proc if file is available

On Linux, the hostname can be set via this file if available, precluding
the need for a `hostname` binary.

Signed-off-by: Dan McGee <dan@archlinux.org>
---
 dhcpcd-hooks/30-hostname |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/dhcpcd-hooks/30-hostname b/dhcpcd-hooks/30-hostname
index 65d1a13..af94262 100644
--- a/dhcpcd-hooks/30-hostname
+++ b/dhcpcd-hooks/30-hostname
@@ -21,7 +21,11 @@ need_hostname()
 try_hostname()
 {
 	if valid_domainname "$1"; then
-		hostname "$1"
+		if [ -w /proc/sys/kernel/hostname ]; then
+			echo "$1" > /proc/sys/kernel/hostname
+		else
+			hostname "$1"
+		fi
 	else
 		syslog err "Invalid hostname: $1"
 	fi
-- 
1.7.5.4

